EXT_PRAYTRACE.md

```diff
@@ -0,0 +1,258 @@
+Domain-Specific Naming Conventions for Photon Raytracing Codes
+===========================================================================
+
+openPMD extension name: `PRAYTRACE`
+
+
+Introduction
+------------
+
+This extension to the openPMD addresses the domain-specific requirements
+met in raytracing codes (e.g. OASYS). Each ray has the information stored
+in its own record, e.g. data/1/particles/rays/0...N/positions and therefore
+has to have the same length.
+
+Particle Based Records (Fields)
+---------------------------
+
+### Root attributes
+
+- `openPMD`: `0.0.2`
+- `openPMDextension`: `PRAYTRACE`
+- `author`: `Aljosa Hafner <aljosa.hafner@ceric-eric.eu>`
+- `date`: `10.10.2019 00:00:00 +0100`
+- `nRays`
+	- type: *(uintX)*
+	- description: Number of initialised rays
+	- scope: *required*
+	- unitSI: 1.0
+	- unitDimension: [unitless] (0., 0., 0., 0., 0., 0., 0.)
+
+### Structure
+data/1/particlesPath/rays/0...N/position...
+
+### Attributes for the individual ray in 'particlesPath/rays/0...N/'
+
+	- `position/`
+		- `position/x`
+			- type: *(floatX)*
+			- description: *x-* coordinate
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [m] (1., 0., 0., 0., 0., 0., 0.)
+		- `position/y`
+			- type: *(floatX)*
+			- description: *y-* coordinate
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [m] (1., 0., 0., 0., 0., 0., 0.)
+		- `position/z`
+			- type: *(floatX)*
+			- description: *z-* coordinates
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [m] (1., 0., 0., 0., 0., 0., 0.)
+	
+	- `direction/'
+		- `direction/x`
+			- type: *(floatX)*
+			- description: *x-* direction
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [m] (1., 0., 0., 0., 0., 0., 0.)
+		- `direction/y`
+			- type: *(floatX)*
+			- description: *x-* direction
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [m] (1., 0., 0., 0., 0., 0., 0.)
+		- `direction/z`
+			- type: *(floatX)*
+			- description: *x-* direction
+			- scope: *required*
+			- unitSI: 1.0
+			- unitDimension: [m] (1., 0., 0., 0., 0., 0., 0.)
+
+	- `eFieldSPolarisation/`
+		- `eFieldSPolarisation/x`
+			- type: *(complexX)*
```

Comment:
why complexX? is float not good enough?
a more descriptive name may be "S polarized E field, x component"

Reply:
You're right...